### DISCUSSION TITLE
consider using mason-* tools to automate installs as much as possible

### DISCUSSION THREAD

**Discussion by andreacfromtheapp** [wojciech-kulik/xcodebuild.nvim#12] [Ideas]
Hi @wojciech-kulik 

I hope you don't mind me suggesting more stuff.

Currently `codelldb` has to be manually installed, along with other requirements. While some of them are manual by necessity, at the least the `dap` bit can be unburdened from the final users with [mason-nvim-dap](https://github.com/jay-babu/mason-nvim-dap.nvim). Something along these is all it would take:

```
 {
    "jay-babu/mason-nvim-dap.nvim",
    opts = function(_, opts) opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "codelldb") end,
  },
```

I hope this helps :)

ps: I'm not sure if more could be done similar to this, and I yet have to understand your tooling before I suggest using more mason, null-ls, and the likes.

**Comment by wojciech-kulik** [comment]
Hi,

Unfortunately, it's not enough, because we'd still have to pass `--liblldb` parameter to `codelldb`. Also, iOS development requires a custom configuration to attach the debugger.

In general, DAP is not a part of this plugin, so it's up to the user if they want to rely on `mason` + `mason-nvim-dap` or set it up manually. The README just provides guidelines as it's not obvious how to make it work.

**Reply by andreacfromtheapp** [reply]
understood :) I tried :P
